daemon/api_base_test.go

```diff
@@ -216,6 +217,7 @@ func (s *apiBaseSuite) SetUpTest(c *check.C) {
 	s.vars = nil
 	s.user = nil
 	s.d = nil
+	s.ctx = nil
```

Comment:
Previously this was done as part of tearing down the test, rather than setting it up... I'm not that familiar with the test suite interface, but are these set as part of the `dirs.SetRootDir()` process? If so, should the cleanup function contain something like this instead?

```go
s.AddCleanup(func() {
        s.d = nil
        s.ctx = nil
        dirs.SetRootDir("")
})
```

Or does the "cleanup" referred to by `AddCleanup()` do these extra bits automatically?

Reply:
I think the idea is that it just needs to be done before another test starts. That could either be at the end of the previous test, or at the start of the current test.